src/zst_decompressor.ts

```diff
@@ -0,0 +1,237 @@
+ *
+ * @param {string} file - The name of the file to decompress and analyze.
+ * @param {number} [start=0] - The starting index for decompression.
+ * @returns {Promise} A promise that resolves when the decompression and analysis is complete.
+ */
+const decompressAndAnalyze = async (file, start = 0) => {
+  let these_chunks_counter = 0; // Initialize the chunk counter
+  let file_counter = 1; // Initialize the file counter
+  let total_chunk_counter = 0;
+  const filesProduced = new Set();
+
+  //   const base_path = `/Users/bennyrubanov/Coding_Projects/chessanalysis/data/${file.replace(
+  const base_path = `${__dirname}/../../data/${file.replace('.zst', '')}`;
+
+  // Create a new file path
+  const newFilePath = `${base_path}_${randomUUID()}`;
+  filesProduced.add(newFilePath);
+
+  // Create a new writable strxeam
+  console.log(`Creating file #${file_counter} at ${newFilePath}`);
+  let decompressedStream = fs.createWriteStream(newFilePath, { flags: 'a' });
+
+  // Check if file already exists
+  if (fs.existsSync(newFilePath)) {
+    const stats = fs.statSync(newFilePath);
+    start = stats.size;
+  }
+
+  try {
+    await new Promise<void>((resolve, reject) => {
+      console.log(
+        `Starting decompression of chunk number ${total_chunk_counter}.`
+      );
+
+      let startTime = Date.now();
+
+      // https://www.npmjs.com/package/node-zstandard#decompressionstreamfromfile-inputfile-callback
+      zstd.decompressionStreamFromFile(
+        `${__dirname}/../../data/${file}`,
+        (err, result) => {
+          if (err) return reject(err);
+
+          let fileLength = 0;
+          let batch_files_total_decompressed_size = 0;
+          let analysisPromises = [];
+          let filesBeingAnalyzed = new Set();
+
+          result.on('error', (err) => {
+            return reject(err);
+          });
+
+          result.on('data', async (data) => {
+            decompressedStream.write(data);
+
+            const duration = Date.now() - startTime;
+            const durationFormatted = formatDuration(duration);
+            fileLength += data.length;
+            batch_files_total_decompressed_size += data.length;
+            these_chunks_counter++;
+
+            // Check if the file size exceeds the limit, if so we need to make a new file
+            if (getFileSize(newFilePath) >= SIZE_LIMIT) {
+              console.log(
+                `Finished decompression of data starting from byte ${start} and ending on byte ${
+                  start + fileLength
+                } of ${file} in ${durationFormatted}`
+              );
+              console.log(
+                `Total number of chunks decompressed so far: ${total_chunk_counter}`
+              );
+              // Increment the file counter
+              file_counter++;
+
+              // Create a new file path
+              const newFilePath = `${base_path}_${randomUUID()}`;
+              filesProduced.add(newFilePath);
```

Comment:
what's the point of this variable? doesn't seem to be used

Reply:
legacy stuff I think